source/opt/trim_capabilities_pass.h

```diff
@@ -0,0 +1,100 @@
+// Copyright (c) 2023 Google Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef SOURCE_OPT_TRIM_CAPABILITIES_PASS_H_
+#define SOURCE_OPT_TRIM_CAPABILITIES_PASS_H_
+
+#include <algorithm>
+#include <functional>
+#include <unordered_map>
+#include <unordered_set>
+
+#include "source/opt/ir_context.h"
+#include "source/opt/module.h"
+#include "source/opt/pass.h"
+
+namespace spvtools {
+namespace opt {
+
+// This is required for NDK build. The unordered_set/unordered_map
+// implementation don't work with class enums.
+struct UnsignedHash {
+  unsigned operator()(spv::Capability value) const {
+    return static_cast<unsigned>(value);
```

Comment:
What is the point of having a hash table if the value is not hashed? This deserves a comment to explain why you are doing this.

Reply:
Forgot to call std::hash, the goal here is to provide a hash function for class enums. C++17 handles that, but NDK lags slightly behind on those.
Fixed this code.